main.rb

```diff
@@ -0,0 +1,120 @@
+module Enumerable
+  # each
+  def my_each
+    each do |i|
+      yield i
+    end
+  end
+
+  # each_with_index
+  def my_each_with_index
+    result = map do |element, index|
+      yield "#{index}: #{element} "
+    end
+    result
+  end
+
+  # select
+  def my_select
+    result = []
+    each do |i|
+      result << i if yield(i)
+    end
+    result
+  end
+
+  # all
+  def my_all?
+    my_each do |_elememt|
+      return false unless yield(element)
+    end
+    true
+  end
```

Comment:
- [ ] `my_all?` returns true if the block never returns false or nil
`%w[ant bear cat].all? { |word| word.length >= 3 }` returns true

- [ ] `my_all?` when no block or argument is given returns true when none of the collection members are false or nil
`[4, 5, 6, true, 2].all?` returns true

- [ ] `my_all?` when a class is passed as an argument returns true if all of the collection is a member of such class
`[4, 5, 6, 2, 9].all?(Numeric)` returns true

- [ ] `my_all?` when a Regex is passed as an argument returns true if all of the collection matches the Regex
`p ['d','d'].my_all?(/d/)` should return true

- [ ] `my_all?` when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern
`p [3,3,3,3].my_all?(3)` should return true